lasagne/tests/test_objectives.py

```diff
@@ -1,65 +1,210 @@
-    # keyword arguments.
-    input_layer.get_output.assert_called_with(input, arg1, kwarg1=kwarg1)
-    network_output = input_layer.get_output.return_value
-
-    # The `network_output` and `target` are fed into the loss
-    # function:
-    loss_function.assert_called_with(network_output, target)
-    assert result == loss_function.return_value
-
-
-def test_objective_no_target():
-    from lasagne.objectives import Objective
-
-    input_layer = mock.Mock()
-    loss_function = mock.Mock()
-    input = object()
-    objective = Objective(input_layer, loss_function)
-    result = objective.get_loss(input)
-
-    input_layer.get_output.assert_called_with(input)
-    network_output = input_layer.get_output.return_value
-
-    loss_function.assert_called_with(network_output, objective.target_var)
-    assert result == loss_function.return_value
+        mask_1hot = np.array([0, 1, 1])
+
+        # Multinomial NLL sum is 1 + 2 + 3 = 6
+        # Mean is 2
+        result = self.get_loss(categorical_crossentropy, output, target_1hot,
+                               aggregation='mean')
+        assert result.eval() == 2
+        result = self.get_loss(categorical_crossentropy, output, target_1hot,
+                               aggregation='sum')
+        assert result.eval() == 6
+        # Multinomial NLL sum is (0*0 + 1*0 + 1*1) + (2*0 + 2*1 + 0*0)
+        # + (3*0 + 0*0 + 3*1) = 6
+        # Mean is 2
+        result = self.get_loss(categorical_crossentropy, output, target_2d,
+                               aggregation='mean')
+        assert result.eval() == 2
+        result = self.get_loss(categorical_crossentropy, output, target_2d,
+                               aggregation='sum')
+        assert result.eval() == 6
+
+        # Masked NLL sum is 2 + 3 = 5
+        result_with_mask = self.get_masked_loss(categorical_crossentropy,
+                                                output, target_1hot,
+                                                mask_1hot,
+                                                aggregation='sum')
+        assert result_with_mask.eval() == 5
+
+        # Masked NLL sum is 2 + 3 = 5
+        result_with_mask = self.get_masked_loss(categorical_crossentropy,
+                                                output, target_2d, mask_1hot,
+                                                aggregation='mean')
+        assert abs(result_with_mask.eval() - 5.0/3.0) < 1.0e-9
+
+        # Masked NLL sum is 2 + 3 = 5
+        result_with_mask = self.get_masked_loss(categorical_crossentropy,
+                                                output, target_2d, mask_1hot,
+                                                aggregation='normalized_sum')
+        assert result_with_mask.eval() == 5.0/2.0
+
+    def test_objective(self):
+        from lasagne.objectives import Objective
+
+        input_layer = mock.Mock()
+        loss_function = mock.Mock()
+        input, target, arg1, kwarg1 = (object(),) * 4
```

Comment:
This will give you four identical objects, which isn't the intention here, and makes the tests below that `assert_called_with` much weaker.